### PR TITLE
[Keyboard] Add a new keyboard "Lunakey Pico"

### PR DESCRIPTION
**Pull request by yoichiro**
This pull request adds a new keyboard designed by @yoichiro to the QMK Firmware.

## Description

I sent this pull request to add a new keyboard named "Lunakey Pico" to the QMK Firmware. I created both the simple default key mappings definition and the key mappings definition for VIA.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yoichiro/lunakey_pico/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# This file intentionally left blank
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/rules.mk:1]:
```suggestion
```

File: keyboards/yoichiro/lunakey_pico/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1 @@
+# This file intentionally left blank
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/keymaps/default/rules.mk:1]: This file should be removed

File: keyboards/yoichiro/lunakey_pico/config.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#define WS2812_PIO_USE_PIO1
+
+#define RGBLIGHT_LED_MAP { 0, 1, 2, 3, 4, 5, \
+                           11, 10, 9, 8, 7, 6 }
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+//#define MATRIX_HAS_GHOST
+
+#define TAPPING_TERM 500
+#define PERMISSIVE_HOLD
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/config.h:25]: We generally ask this to be defined at the keymap level instead
```suggestion
```

File: keyboards/yoichiro/lunakey_pico/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______,                   _______, _______, _______, _______, _______, _______,
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+                              KC_LGUI, KC_LALT, LOWER,   KC_SPC,  KC_ENT,  RAISE,   KC_RALT, KC_RGUI
+//+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+--------+
+  )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/keymaps/via/keymap.c:69]:
```suggestion
```

File: keyboards/yoichiro/lunakey_pico/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+enum layer_number {
+  _QWERTY = 0,
+  _LOWER,
+  _RAISE,
+  _ADJUST,
+};
+
+#define LOWER MO(_LOWER)
+#define RAISE MO(_RAISE)
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/keymaps/via/keymap.c:14]:
```suggestion
#define LOWER FN_MO13
#define RAISE FN_MO23
```

File: keyboards/yoichiro/lunakey_pico/lunakey_pico.c [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+#include "lunakey_pico.h"
+
+//void keyboard_post_init_kb(void) {
+//    debug_enable   = true;
+//    debug_matrix   = true;
+//    debug_keyboard = true;
+//    debug_mouse    = true;
+//    keyboard_post_init_user();
+//}
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/lunakey_pico.c:12]:
```suggestion
```

File: keyboards/yoichiro/lunakey_pico/mcuconf.h [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+#undef RP_SIO_USE_UART0
+#define RP_SIO_USE_UART0 TRUE
+#undef RP_SIO_USE_UART1
+#define RP_SIO_USE_UART1 FALSE
+
+//#undef RP_SERIAL_USE_UART0
+//#define RP_SERIAL_USE_UART0 TRUE
+//#undef RP_SERIAL_USE_UART1
+//#define RP_SERIAL_USE_UART1 FALSE
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/mcuconf.h:16]:
```suggestion
```

File: keyboards/yoichiro/lunakey_pico/halconf.h [hunk 8]
```diff
@@ -0,0 +1,9 @@
+// Copyright 2022 Yoichiro Tanaka (@yoichiro)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define HAL_USE_SIO TRUE
+//#define HAL_USE_SERIAL TRUE
```
Comment (zvecr) [inline keyboards/yoichiro/lunakey_pico/halconf.h:7]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/yoichiro/lunakey_pico/keymaps/default/rules.mk (hunk 2 above)
Comment (yoichiro) [reply keyboards/yoichiro/lunakey_pico/keymaps/default/rules.mk]: I removed this file at f532758.

**Comment by yoichiro** [comment]
@zvecr Thank you for many suggestions! Could you review them again?

File: keyboards/yoichiro/lunakey_pico/info.json [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        "vid": "0x5954"
+    },
+    "rgblight": {
+        "led_count": 12,
+        "pin": "GP6",
+        "sleep": true,
+        "split": true,
+        "split_count": [6, 6],
+        "animations": {
+            "all": true
```
Comment (drashna) [inline keyboards/yoichiro/lunakey_pico/info.json:35]: Could you explicitly define the animations:
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": false,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by yoichiro** [comment]
@drashna Thank you the suggestion! I committed your suggestion just now. Could you review this again?

**Review by drashna — APPROVED** [review]

File: keyboards/yoichiro/lunakey_pico/readme.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# Lunakey Pico
+
+![lunakey_pico](https://i.imgur.com/S5N8AFh.jpg)
```
Comment (fauxpark) [inline keyboards/yoichiro/lunakey_pico/readme.md:3]:
```suggestion
![lunakey_pico](https://i.imgur.com/S5N8AFhh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by yoichiro** [comment]
@fauxpark Thank you for the suggestion! I committed the suggestion just now.

**Comment by yoichiro** [comment]
@drashna @fauxpark @zvecr Is there something else I should do?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**